src/handler.ts

```diff
@@ -104,9 +104,10 @@ export default class DataHandler {
       const { options: { name } } = this.operators[i];
 
       try {
-        // if the data is null, it is a signal to stop processing
+        // if the data is null or empty, it is a signal to stop processing
         // this can happen if an upstream handler needed to prevent a downstream operator
-        if (handledData === null) {
+        // and we don't want to submit empty events to the configured destination
+        if (handledData === null || DataHandler.isEmptyData(handledData)) {
```

Comment:
Why isn't this check needed for line 120? I guess I'm not sure if any (specifically the last in the pipeline) `operator.handleData` could ever return empty not null data?

Reply:
On line 120 `handledData` is assumed to be an array, and each item in the array is passed back into `handleData` as the `data` arg. This would eventually hit the check on line 110 again as long as there are additional operators configured after `FanOutOperator`. If there aren't additional operators configured, I don't think we'd need to worry about `null` or `empty` data being sent to `FS.event` since data is sent to the configured destination through a `FunctionOperator` at the end of the operator chain.

Reply:
Oh okay, I missed the fact that the `FunctionOperator` should always be the last in the the operator chain(`this.operators`), good to know! Thx!

Reply:
This is tough to know without digging into the `DataLayerObserver` initialization code. I just came across this recently: https://github.com/fullstorydev/fullstory-data-layer-observer/blob/11c4623525585fdb799d0f753beafc362b6f2550/src/observer.ts#L207